core/src/test/java/hudson/FilePathTest.java

```diff
@@ -736,4 +737,38 @@ public void testEOFbrokenFlush() throws IOException, InterruptedException {
         // and now fail when flush is bad!
         tmpDirPath.child("../" + archive.getName()).untar(outDir, TarCompression.NONE);
     }
+
+    @Test public void deleteRecursive() throws Exception {
```

Comment:
Just a note, https://bugs.openjdk.java.net/browse/JDK-8069345 prevents us from using `Files.walkFileTree` to implement recursive deletion. :(